integrations/azure-devops/azure_devops/http/http_client.py

```diff
@@ -0,0 +1,150 @@
+import httpx
+from httpx import Timeout, BasicAuth, Response
+from typing import Any, AsyncGenerator, Optional
+from port_ocean.utils import http_async_client
+from loguru import logger
+
+PAGE_SIZE = 50
+REQUEST_TIMEOUT_SECONDS = 30
+CONTINUATION_TOKEN_HEADER = "x-ms-continuationtoken"
+ALLOW_REDIRECTS = True
+
+
+class HTTPBaseClient:
+    def __init__(self, personal_access_token: str) -> None:
+        self._async_client = http_async_client
+        # Username isn't required in basic auth to Azure Devops
+        self._auth = BasicAuth("", personal_access_token)
+        self._timeout = Timeout(REQUEST_TIMEOUT_SECONDS)
+        self._follow_redirects = ALLOW_REDIRECTS
+
+    @staticmethod
+    def _parse_response_values(response: Response) -> list[dict[Any, Any]]:
+        return response.json()["value"]
+
+    def send_sync_get_request(
+        self, url: str, params: Optional[dict[str, Any]] = None
+    ) -> Response:
+        try:
+            response = httpx.get(
+                url,
+                params=params,
+                auth=self._auth,
+                timeout=self._timeout,
+                follow_redirects=self._follow_redirects,
+            )
+            response.raise_for_status()
+        except httpx.HTTPStatusError as e:
+            if response.status_code == 401:
+                logger.error(
+                    f"Couldn't access url {url} . Make sure the PAT (Personal Access Token) is valid!"
+                )
+            else:
+                logger.debug(
+                    f"Request with bad status code {response.status_code}: GET to url {url}: {str(e)}"
+                )
+                raise e
+        except Exception as e:
+            logger.error(f"Couldn't send get request to url {url}: {str(e)}")
+            raise e
+        return response
```

Comment:
why would you need this?

Reply:
This is used to get an item from a repository. This is sync because the current implementation of the _search in the EntityProcessors is sync. I've tried:
1. To use asyncio.run
2. Do something like this: 
  `loop = asyncio.new_event_loop()
    asyncio.set_event_loop(loop)
    a = loop.run_until_complete(..)`
But I didn't manage to run the async method synchronously.

Reply:
lets try to solve it together